pkg/metrics/metrics.go

```diff
@@ -582,12 +586,49 @@ func prometheusToken(c client.Client) (*string, error) {
 	}
 
 	if len(token) == 0 {
-		return nil, fmt.Errorf("failed to find token secret for prometheus-k8s SA")
+		// If serviceaccount token doesn't exist, try to use token request to get a token
+		token, err = RequestPrometheusServiceAccountAPIToken()
+		if err != nil {
+			return nil, fmt.Errorf("failed to find token secret for prometheus-k8s SA due to %v", err)
+		}
 	}
 
 	return &token, nil
 }
 
+func NewKubernetesClient() (*kubernetes.Clientset, error) {
+	config, err := rest.InClusterConfig()
```

Comment:
Can we have a look at https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.17.0/pkg/client#SubResourceClientConstructor.SubResource to leverage controller-runtime client to make the call to create the token?

Reply:
Updated to use controller runtime